SU2_CFD/src/solvers/CScalarSolver.cpp

```diff
@@ -0,0 +1,958 @@
+/*!
+ * \file CScalarSolver.cpp
+ * \brief Main subrotuines of CScalarSolver class
+ * \author F. Palacios, A. Bueno
+ * \version 7.1.1 "Blackbird"
+ *
+ * SU2 Project Website: https://su2code.github.io
+ *
+ * The SU2 Project is maintained by the SU2 Foundation
+ * (http://su2foundation.org)
+ *
+ * Copyright 2012-2021, SU2 Contributors (cf. AUTHORS.md)
+ *
+ * SU2 is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "../../include/solvers/CScalarSolver.hpp"
+#include "../../../Common/include/parallelization/omp_structure.hpp"
+#include "../../../Common/include/toolboxes/geometry_toolbox.hpp"
+
+
+CScalarSolver::CScalarSolver(void) : CSolver() { }
+
+CScalarSolver::CScalarSolver(CGeometry* geometry, CConfig *config) : CSolver() {
+
+  nMarker = config->GetnMarker_All();
+
+  /*--- Store the number of vertices on each marker for deallocation later ---*/
+  nVertex.resize(nMarker);
+  for (unsigned long iMarker = 0; iMarker < nMarker; iMarker++)
+    nVertex[iMarker] = geometry->nVertex[iMarker];
+
+  /* A grid is defined as dynamic if there's rigid grid movement or grid deformation AND the problem is time domain */
+  dynamic_grid = config->GetDynamic_Grid();
+
+#ifdef HAVE_OMP
+  /*--- Get the edge coloring, see notes in CEulerSolver's constructor. ---*/
+  su2double parallelEff = 1.0;
+  const auto& coloring = geometry->GetEdgeColoring(&parallelEff);
+
+  ReducerStrategy = parallelEff < COLORING_EFF_THRESH;
+
+  if (ReducerStrategy && (coloring.getOuterSize()>1))
+    geometry->SetNaturalEdgeColoring();
+
+  if (!coloring.empty()) {
+    auto groupSize = ReducerStrategy? 1ul : geometry->GetEdgeColorGroupSize();
+    auto nColor = coloring.getOuterSize();
+    EdgeColoring.reserve(nColor);
+
+    for(auto iColor = 0ul; iColor < nColor; ++iColor)
+      EdgeColoring.emplace_back(coloring.innerIdx(iColor), coloring.getNumNonZeros(iColor), groupSize);
+  }
+
+  nPoint = geometry->GetnPoint();
+  omp_chunk_size = computeStaticChunkSize(nPoint, omp_get_max_threads(), OMP_MAX_SIZE);
+#else
+  EdgeColoring[0] = DummyGridColor<>(geometry->GetnEdge());
+#endif
+}
+
+CScalarSolver::~CScalarSolver() {
+
+  delete nodes;
+
+}
+
```

Comment:
Indeed that is annoying, I think I've put su2double in front of all flow solver vars because of that. Otherwise the problem comes back, or we have to disable clang-format around certain sections.

Reply:
> The 1 space indentation of protected seems weird as well

It's the price to pay for peace of mind with formatting, it's easier to get used to that than to every personal style in SU2 :laughing: